src/main/java/uk/nhs/digital/nhsconnect/nhais/repository/SchedulerTimestampRepositoryExtensions.java

```diff
@@ -0,0 +1,30 @@
+package uk.nhs.digital.nhsconnect.nhais.repository;
+
+import java.time.LocalDateTime;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+public interface SchedulerTimestampRepositoryExtensions {
+    boolean updateTimestamp(
+        SchedulerTimestampRepositoryExtensions.UpdateTimestampParams updateTimestampParams,
+        SchedulerTimestampRepositoryExtensions.UpdateTimestampDetails updateTimestampDetails);
+
+    @RequiredArgsConstructor
+    @Getter
+    @ToString
+    class UpdateTimestampParams {
+        private final String schedulerType;
+        private final long minutes;
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    @ToString
+    class UpdateTimestampDetails {
+        private final String schedulerType;
+        private final LocalDateTime timestamp;
+    }
```

Comment:
Why two classes instead of one?
`schedulerType` is constant in current implementation (if it's needed for future then leave it)

Reply:
Are those classes even needed? You can define `public boolean updateTimestamp(String schedulerType, LocalDateTime timestamp, long minutes) {...}` and get rid of them

Reply:
scheduler Type configuration is needed for future, just to make this repository reusable for other purposes

Reply:
I took example from the outboundState solution, jest I can simplify it